Lib/ufo2ft/util.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import print_function, division, absolute_import
+
+
+def makeOfficialGlyphOrder(font, glyphOrder=None):
+    """ Make the final glyph order for 'font'.
+
+    If glyphOrder is None, try getting the font.glyphOrder list.
+    If not explicit glyphOrder is defined, sort glyphs alphabetically.
+
+    If ".notdef" glyph is present in the font, force this to always be
+    the first glyph (at index 0).
+    """
+    if glyphOrder is None:
+        glyphOrder = getattr(font, "glyphOrder", ())
+    orderedGlyphs = [".notdef"] if ".notdef" in font else []
+    for glyphName in glyphOrder:
+        if glyphName == ".notdef":
+            continue
+        if glyphName not in font:
+            continue
+        orderedGlyphs.append(glyphName)
+    orderedGlyphSet = set(orderedGlyphs)
+    for glyphName in sorted(font.keys()):
+        if glyphName not in orderedGlyphSet:
+            orderedGlyphs.append(glyphName)
+    return orderedGlyphs
```

Comment:
How about:

~~~python
names = set(font.keys())
order = []
if ".notdef" in names:
    names.remove(".notdef")
    order.append(".notdef")
for name in glyphOrder:
    if name not in names:
        continue
    names.remove(name)
    order.append(name)
for name in sorted(names):
    order.append(name)
~~~

This way you first create a set of all glyphs which you remove from as you go, at the end you don't need to sort all the glyphs in the font.

Reply:
thanks, I like that